File: MtgCardOrganizer.Core/Entities/Decks/DeckCard.cs
using System.Collections.Generic;
using MtgCardOrganizer.Core.Entities.Cards;

namespace MtgCardOrganizer.Core.Entities.Decks
{
    public class DeckCard : Entity
    {
        public int DeckId { get; set; }
        public int CardId { get; set; }

        public Deck Deck { get; set; }
        public Card Card { get; set; }

        public DeckPart Part { get; set; }
        public int Count { get; set; }
    }
}

File: mtg-card-organizer/src/app/deck/models/deck-card.ts
import { Card } from '../../card/models/card';
import { DeckPart } from './deck-part';

export class DeckCard {
  cardId: number;
  card: Card;
  part: DeckPart;
  count: number;

  constructor(init?: Partial<DeckCard>) {
    Object.assign(this, init);
  }
}
